src/Core/IdentityServer/ResourceOwnerPasswordValidator.cs

```diff
@@ -60,25 +60,32 @@ public async Task ValidateAsync(ResourceOwnerPasswordValidationContext context)
             //    return;
             //}
 
-            if (_captchaValidationService.ServiceEnabled && _currentContext.IsBot)
+            await ValidateAsync(context, context.Request);
```

Comment:
So this will now validate username/password before captcha validation?

Reply:
I haven't tested whether there's a detectable difference between successful validations and unsuccessful, but I think that's a valid concern here.

I think I need to rethink this early validation idea